crates/symbolicator-service/src/services/sourcemap_lookup.rs

```diff
@@ -579,6 +552,135 @@ impl ArtifactFetcher {
             sourcemap_url: sourcemap_url.map(Arc::new),
         })
     }
+
+    /// Queries the Sentry API for a single file (by its [`DebugId`] and file stem).
+    async fn query_sentry_for_file(&mut self, key: &FileKey) {
+        let mut debug_ids = BTreeSet::new();
+        if let Some(debug_id) = key.debug_id() {
+            debug_ids.insert(debug_id);
+        }
+
+        let mut file_stems = BTreeSet::new();
+        if let Some(url) = key.abs_path() {
+            let stem = extract_file_stem(url);
+            file_stems.insert(stem);
+        }
+
+        self.query_sentry_for_files(debug_ids, file_stems).await
+    }
+
+    /// Queries the Sentry API for a bunch of [`DebugId`]s and file stems.
+    ///
+    /// This will also download all referenced [`ArtifactBundle`]s directly and persist them for
+    /// later access.
+    /// Individual files are not eagerly downloaded, but their metadata will be available.
+    async fn query_sentry_for_files(
+        &mut self,
+        debug_ids: BTreeSet<DebugId>,
+        file_stems: BTreeSet<&str>,
+    ) {
+        // TODO(sourcemap): this is where we want to hook in the not-yet-existing Sentry API
+        let _ = (debug_ids, file_stems);
+        #[allow(unused)]
+        enum FileResult {
+            Individual(SearchArtifactResult),
+            ArtifactBundle(String),
```

Comment:
Should this variant contain a `Url`?

Reply:
Thats also possible. I was thinking about the whole response being thrown away for a single invalid url, (and thinking about parsing it when we have already seen it previously; but thats a case of premature optimization probably).

Considering that this should be an unexpected error, it is also okay to just parse this as part of the response.